torch/_inductor/codegen/cpp.py

```diff
@@ -931,22 +931,28 @@ def load(self, name: str, index: sympy.Expr):
         expanded_index = sympy.expand(index)
         new_index = self.scale_index_with_offset(index, self.tiling_factor)
 
-        if expanded_index == new_index:
-            line = f"at::vec::Vectorized<float>({var}[{cexpr(index)}])"
-        else:
-            if V.graph.get_dtype(name) in [torch.bool, torch.uint8]:
-                nelements = codecache.pick_vec_isa().nelements()
-                if var not in self.var_vec_buf_map:
-                    self.var_vec_buf_map[var] = f"g_tmp_buffer_{var}"
-                    self.loads.writeline(
-                        f"float {self.var_vec_buf_map[var]}[{nelements}] = {{0}};"
-                    )
+        related_to_inner_most_index = expanded_index != new_index
+        if V.graph.get_dtype(name) in [torch.bool, torch.uint8]:
+            nelements = codecache.pick_vec_isa().nelements()
+            if var not in self.var_vec_buf_map:
+                self.var_vec_buf_map[var] = f"g_tmp_buffer_{var}"
                 self.loads.writeline(
-                    f"flag_to_float({var} + {cexpr(new_index)}, {self.var_vec_buf_map[var]}, {nelements});"
+                    f"float {self.var_vec_buf_map[var]}[{nelements}] = {{0}};"
                 )
-                line = f"at::vec::Vectorized<float>::loadu({self.var_vec_buf_map[var]})"
-            else:
+            flag_to_float_func_name = (
+                "flag_to_float"
+                if related_to_inner_most_index
+                else "flag_to_float_broadcast"
```

Comment:
We needn't provide a dedicated function to broadcast a scalar value. The code could be as follow.
```python
f"at::vec::Vectorized<float>({var}[{cexpr(index)}] ? 0xFFFFFFFF : 0)"
```

Reply:
With `f"at::vec::Vectorized<float>({var}[{cexpr(index)}] ? 0xFFFFFFFF : 0)"`, mask `True` would be `4.29497e+09` instead of `-nan`, which leads to the wrong result.